check_api_key should not return HTTP401

Azher2Ali opened the pull request:
In order to return messages a boolean property "valid".
When the apikey is invalid, return HTTP 200 with response body like:

{
  "valid": false,
  "error": "...message explaining why invalid"
}

 **_Expected Output_** 
<img width="1700" alt="Screenshot 2023-04-03 at 12 34 19 PM" src="https://user-images.githubusercontent.com/121898125/229572296-c68d3b5b-846c-4283-9adf-11aec7334304.png">

Azher2Ali committed:
Added a Global Exceptional Handler for 401 request 

Added a Global Exceptional Handler for 401 request

Azher2Ali committed:
Created a custom exception, instead of invalid token

Created a custom exception, instead of invalid token

Azher2Ali committed:
Created a Custom Exception for an invalid token

Created a Custom Exception for an invalid token

joneubank commented on src/main/java/bio/overture/ego/service/TokenService.java:
```diff
@@ -399,16 +399,19 @@ private String getSignedToken(TokenClaims claims) {
     if (apiKey == null) {
       log.debug("Null apiKey");
-      throw new InvalidTokenException("No apiKey field found in POST request");
+      throw new ClientInvalidTokenException("No apiKey field found in POST request");
```
no key provided should be a 400 for invalid request

Azher2Ali replied:
400 means, do you refer to invalid token exception?

joneubank replied:
`InvalidRequestException` I think?

joneubank commented on src/main/java/bio/overture/ego/service/TokenService.java:
```diff
@@ -399,16 +399,19 @@ private String getSignedToken(TokenClaims claims) {
+        findByApiKeyString(apiKey)
+            .orElseThrow(
+                () -> {
+                  return new ClientInvalidTokenException("ApiKey not found");
```
apikey not found should be a 400 for invalid request

joneubank commented on src/main/java/bio/overture/ego/service/TokenService.java:
```diff
@@ -399,16 +399,19 @@ private String getSignedToken(TokenClaims claims) {
 
     if (aK.isRevoked())
-      throw new InvalidTokenException(
+      throw new ClientInvalidTokenException(
```
👍

joneubank reviewed (changes requested):
Need to properly respond when the request is invalid from the user (missing apikey)

Azher2Ali committed:
Update TokenService.java

As per the changes required, the ClientInvalidTokenException is been changed to InvalidTokenException. As it's an invalid request because the required parameter is not provided.

Azher2Ali reviewed (commented):
As per the request , the changes are made

joneubank reviewed (approved):
Thanks!
